DEVELOPING.md

```diff
@@ -1,75 +1,241 @@
+```
 
 
-Building TLC / `tlatools.jar`
-=============================
+Build & Test TLA⁺ Tools
+----------------------
+Install the following dependencies to your path:
+ * [OpenJDK](https://openjdk.org/) version 11+
+ * [Apache Ant](https://ant.apache.org/) version 1.9.8+
+
+Clone this repo & open a shell in its root, then run:
+```bash
+cd tlatools/org.lamport.tlatools
+ant -f customBuild.xml info compile compile-test dist # Builds tla2tools.jar
+java -cp dist/tla2tools.jar tlc2.TLC test-model/pcal/Bakery.tla # Runs an example
+ant -f customBuild.xml test # Runs unit tests
+```
 
-While the Toolbox is maintained in [Eclipse](https://github.com/tlaplus/tlaplus/tree/master/general/ide), sometimes changes to TLC do not require building the entire GUI. In such cases, it can be beneficial to compile TLC on the command line. 
+The unit tests should all succeed.
+The compiled `tla2tools.jar` will be output to `tlatools/org.lamport.tlatools/dist/tla2tools.jar`, and can be used the same as any `tla2tools.jar` you download from the releases.
 
-While `Maven` is used for the Toolbox (see below), `ant` can be used for the command line. It can be installed with `brew` or `apt` with the package name `ant` (e.g. `brew install ant` and `sudo apt-get install ant`).
+Build & Test Toolbox IDE
+------------------------
+*Note: these instructions are likely to fail and must be fixed; see [this issue](https://github.com/tlaplus/tlaplus/issues/851).*
 
-Make sure that you're using Java 11+. If you're using a lower version you'll see compilation errors.
+Install the following dependencies to your path:
+ * [OpenJDK](https://openjdk.org/) version 11+
+ * [Apache Ant](https://ant.apache.org/) version 1.9.8+
+ * [Apache Maven](https://maven.apache.org/) version 3.9.7+
 
-You can build TLC from a fresh clone of this repository+:
+Clone this repo & open a shell in its root, then run:
 ```bash
-$ git clone https://github.com/tlaplus.git
-$ cd tlaplus/tlatools/org.lamport.tlatools
-$ ant -f customBuild.xml info clean compile compile-test dist
+mvn install -Dmaven.test.skip=true # Builds Toolbox
+mvn verify # Runs tests
+```
+
+The Toolbox distributables will be output in the `toolbox/org.lamport.tla.toolbox.product.product/target/products` directory.
+
+Developing with the Eclipse IDE
+-------------------------------
+
+You are free to use your own editor, but this project is largely developed in the [Eclipse](https://www.eclipse.org/) integrated development environment (IDE).
+
+### Developing the TLA⁺ Tools in Eclipse
+
+To develop the TLA⁺ Tools in Eclipse, use Eclipse to open the `tlatools/org.lamport.tlatools` project directory.
+Within that directory, the Eclipse project configuration lives in the `.classpath` and `.project` files.
+The `.classpath` file is also consumed by the Toolbox IDE build, so some local modifications are required before Eclipse can fully build & test the project.
+Copy & paste the following into a file called `classpath.patch` within your repo:
```

Comment:
It does not, Eclipse cannot find any of the test dependencies.

Reply:
With what error message does the Oomph setup fail?

Reply:
I don't use the oomph setup at all for developing the TLA+ tools (not the toolbox). I just open the tlatools/org.lamport.tlatools project.

Reply:
We should get there eventually but that's not supported right now. Thus,  it shouldn't be in DEVELOPING.md.

Reply:
I think having people be able to start development by opening a single directory in eclipse is much simpler than running the long and failure-prone oomph process, especially since we want to focus development on the tools.

Reply:
See https://github.com/tlaplus/tlaplus/pull/938#discussion_r1644990978